pkg/controllers/dynakube/extension/eec/statefulset.go

```diff
@@ -370,3 +360,22 @@ func setVolumes(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
 		}
 	}
 }
+
+func setPersistentVolumeClaim(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
+	return func(o *appsv1.StatefulSet) {
+		if dk.Spec.Templates.ExtensionExecutionController.PersistentVolumeClaim != nil {
```

Comment:
since it's a slice you override original one. So you can try to create a unit test and set `spec.VolumeClaimTemplates` with some value and then it will gone.

Reply:
Dynakube CRD allows to define single PVC spec and we want exactly one PVC object in EEC statefulset if dk.PVC spec is provided. Fresh StatefulSet has no PVC. 
There is no need to use `append` now. 

But I forgot to update TestPersistentVolumeClaimRetentionPolicy and I will add  TestPersistentVolumeClaim test to check if dk.PVC and statefulset.PVC definitions are the same.

Reply:
thanks fro clarification.